File: internal/content/testdata/bench/csharp/nc-word-ladder.cs
using System.Collections.Generic;

public class Solution {
    public long ladder_length(string begin_word, string end_word, string[] word_list) {
        var wordSet = new HashSet<string>(word_list);
        if (!wordSet.Contains(end_word)) return 0;
        var queue = new Queue<(string word, long length)>();
        queue.Enqueue((begin_word, 1));
        var visited = new HashSet<string>();
        visited.Add(begin_word);
        while (queue.Count > 0) {
            var (word, length) = queue.Dequeue();
            char[] chars = word.ToCharArray();
            for (int i = 0; i < chars.Length; i++) {
                char orig = chars[i];
                for (char c = 'a'; c <= 'z'; c++) {
                    chars[i] = c;
                    string candidate = new string(chars);
                    if (candidate == end_word) return length + 1;
                    if (wordSet.Contains(candidate) && !visited.Contains(candidate)) {
                        visited.Add(candidate);
                        queue.Enqueue((candidate, length + 1));
                    }
                }
                chars[i] = orig;
            }
        }
        return 0;
    }
}

File: internal/content/testdata/bench/typescript/nc-word-ladder.ts
function ladder_length(begin_word: string, end_word: string, word_list: string[]): number {
    const wordSet = new Set(word_list);
    if (!wordSet.has(end_word)) return 0;
    const queue: [string, number][] = [[begin_word, 1]];
    const visited = new Set<string>([begin_word]);
    let qi = 0;
    while (qi < queue.length) {
        const [word, length] = queue[qi++];
        for (let i = 0; i < word.length; i++) {
            for (let c = 97; c <= 122; c++) {
                const candidate = word.slice(0, i) + String.fromCharCode(c) + word.slice(i + 1);
                if (candidate === end_word) return length + 1;
                if (wordSet.has(candidate) && !visited.has(candidate)) {
                    visited.add(candidate);
                    queue.push([candidate, length + 1]);
                }
            }
        }
    }
    return 0;
}
